runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/util/Attributes.kt

```diff
@@ -65,39 +63,44 @@ public operator fun <T : Any> Attributes.get(key: AttributeKey<T>): T = getOrNul
 /**
  * Removes an attribute with the specified [key] and returns its current value, throws an exception if an attribute doesn't exist
  */
-public fun <T : Any> Attributes.take(key: AttributeKey<T>): T = get(key).also { remove(key) }
+public fun <T : Any> MutableAttributes.take(key: AttributeKey<T>): T = get(key).also { remove(key) }
 
 /**
  * Set a value for [key] only if it is not already set
  */
-public fun <T : Any> Attributes.putIfAbsent(key: AttributeKey<T>, value: T) {
+public fun <T : Any> MutableAttributes.putIfAbsent(key: AttributeKey<T>, value: T) {
     if (!contains(key)) set(key, value)
 }
 
 /**
  * Set a value for [key] only if [value] is not null
  */
-public fun <T : Any> Attributes.setIfValueNotNull(key: AttributeKey<T>, value: T?) {
+public fun <T : Any> MutableAttributes.setIfValueNotNull(key: AttributeKey<T>, value: T?) {
     if (value != null) set(key, value)
 }
 
 /**
  * Removes an attribute with the specified [key] and returns its current value, returns `null` if an attribute doesn't exist
  */
-public fun <T : Any> Attributes.takeOrNull(key: AttributeKey<T>): T? = getOrNull(key).also { remove(key) }
+public fun <T : Any> MutableAttributes.takeOrNull(key: AttributeKey<T>): T? = getOrNull(key).also { remove(key) }
 
 /**
  * Merge another attributes instance into this set of attributes favoring [other]
  */
-public fun Attributes.merge(other: Attributes) {
+public fun MutableAttributes.merge(other: Attributes) {
     other.keys.forEach {
         @Suppress("UNCHECKED_CAST")
         set(it as AttributeKey<Any>, other[it])
     }
 }
 
-private class AttributesImpl : Attributes {
+private class AttributesImpl constructor(seed: Attributes) : MutableAttributes {
```

Comment:
should this be named `MutableAttributesImpl`?

Reply:
I don't know if it matters since it's `private`